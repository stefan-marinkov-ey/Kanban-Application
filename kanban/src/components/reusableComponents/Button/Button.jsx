import React from "react";
import PropTypes from "prop-types";
import { StyledPrimaryButton } from "../../styleComponents/Container/Button_styled";

const Button = ({ label, onClick, type, loading }) => {
  return (
    <StyledPrimaryButton type={type} onClick={onClick} disabled={loading}>
      {label}
    </StyledPrimaryButton>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string,
};

Button.defaultProps = {
  children: "",
  type: "text",
  label: "",
};

export default Button;
