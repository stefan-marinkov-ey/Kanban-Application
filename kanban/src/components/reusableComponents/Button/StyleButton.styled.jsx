import styled from "styled-components";

export const StyleButton = styled.button`
  height: 2rem;
  background: ${(props) => props.background};
  border: 2px solid ${(props) => props.theme.toggleBorder};
  color: ${(props) => props.theme.color};
  border-radius: 30px;
  cursor: pointer;
  font-size: 0.8rem;
  padding: 0.6rem;
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    cursor: pointer;
    border-radius: 5px;
    transition: all 0.5s ease-in-out;
    opacity: 0.4;
    background: ${(props) => props.theme.background};
    color: ${(props) => props.theme.color};
  }
`;
