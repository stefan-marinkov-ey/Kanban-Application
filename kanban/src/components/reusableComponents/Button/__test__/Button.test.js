import React from "react";
import Button from "../Button";
import * as renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

describe("Button render", () => {
  const wrapper = shallow(<Button></Button>);
  expect(wrapper).toBe(true);
  it("should match snapshot", () => expect(wrapper).toMatchSnapshot());
});
