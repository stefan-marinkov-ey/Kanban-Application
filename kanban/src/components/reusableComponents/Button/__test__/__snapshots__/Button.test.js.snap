// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render 1`] = `
Array [
  <styled.button
    className=""
    onMouseDown={[Function]}
    type="text"
  >
    
  </styled.button>,
]
`;

exports[`Button Component should find label 1`] = `Array []`;

exports[`Button Component should render correctly 1`] = `
Object {
  "children": "Some label with text",
  "className": "",
  "disabled": undefined,
  "onClick": undefined,
  "onMouseDown": [Function],
  "type": "text",
}
`;
